README.md

```diff
@@ -1,37 +1,63 @@
 # Eclipse Che on OpenShift
 
+## Table Of Content
+
+* [How to build the RedHat Che distribution](#how-to-build-the-redhat-che-distribution)
+* [How to build the upstream openshift-connector branch for development purposes](#how-to-build-the-upstream-openshift-connector-branch-for-development-purposes)
+* [How to run Che on OpenShift](#how-to-run-che-on-openshift)
+
 ## How to build the RedHat Che distribution
 
+### TL;DR
+
+There is a good chance that you want to build rh-che with the dashboard but without keycloak support (because it's easier to test) and as fast as possible :bowtie::
+
+```
+git clone https://github.com/redhat-developer/rh-che
+cd rh-che
+mvn -DwithoutKeycloak `# disable Keycloak support` \
+    -Pfast            `# skip tests and other verifications` \
+    clean install
+```
+
+If you have already cloned [eclipse/che](https://github.com/eclipse/che) and built [openshift-connector branch](https://github.com/eclipse/che/tree/openshift-connector) you will want to:
+
+```
+UPSTREAM_CHE_PATH=/path/to/upstream/che
+mvn -DwithoutKeycloak                          `# disable Keycloak support` \
+    -Pfast                                     `# skip tests and other verifications` \
+    -DlocalCheRepository=${UPSTREAM_CHE_PATH}  `# get already built upstream Che` \
+    clean install
+```
+
 #### What is the RedHat Che distribution
 
-The RedHat distribution of Eclipse Che is a RedHat-specific packaging of the Che assemblies
-that allows adding some RedHat specific plugins / behaviors upton the standard upstream Che
+The RedHat distribution of Eclipse Che is a RedHat-specific packaging of Che assemblies
+that adds some RedHat specific plugins / behaviors upton the standard upstream Che
```

Comment:
that adds some RedHat specific plugins / behaviors **on top of** the standard upstream Che